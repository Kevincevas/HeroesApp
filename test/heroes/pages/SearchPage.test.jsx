import { fireEvent, render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { SearchPage } from "../../../src/heroes/pages/SearchPage";
import React from "react";

const mockUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockUsedNavigate,
}));

describe('Pruebas en SearchPage', () => {

    beforeEach( () => jest.clearAllMocks() );
    
    test('debe de mostrarse correctamente con valores por defecto', () => {

        const {container} = render(
            <MemoryRouter>
                <SearchPage />
            </MemoryRouter>
        );
        
        expect(container).toMatchSnapshot();
    });

    test('debe de a Batman y el input con el valor del queryString', () => {

        render(
            <MemoryRouter initialEntries={['/search?q=batman']}>
                <SearchPage />
            </MemoryRouter>
        );
        
        const input = screen.getByRole('textbox');
        expect(input.value).toBe('batman');

        const img = screen.getByRole('img');
        expect(img.src).toContain('/heroes/dc-batman.jpg');
    });

    test('debe de mostrar un error si no se encuentra el heroe (batman123)', () => {
        render(
            <MemoryRouter initialEntries={['/search?q=batman123']}>
                <SearchPage />
            </MemoryRouter>
        );

        const alert = screen.getByLabelText('alert-danger');
        //expect(alert).toBe('String');


    });

    test('debe de llamar el navigate a la pantalla nueva', () => {
        render(
            <MemoryRouter initialEntries={['/search?q=batman123']}>
                <SearchPage />
            </MemoryRouter>
        );

        const input = screen.getByRole('textbox');
        //evento para simular llenar un campo o input
        fireEvent.change(input, {target:{name:'searchText', value:'superman'}});
        
        const form = screen.getByRole('form');
        //evento para simular el submit de un formulario
        fireEvent.submit(form);
        
        expect(mockUsedNavigate).toHaveBeenCalledWith('?q=superman')
    })
    
  
})
