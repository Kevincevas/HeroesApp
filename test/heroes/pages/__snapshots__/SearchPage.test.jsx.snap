// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en SearchPage debe de mostrarse correctamente con valores por defecto 1`] = `
<div>
  <h1>
    Search
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form
        aria-label="form"
      >
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search a Hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-2"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Resoults
      </h4>
      <hr />
      <div
        aria-label="alert-info"
        class="alert alert-primary animate__animated animate__backInRight"
      >
         Search a Hero 
        <b />
         
      </div>
    </div>
  </div>
</div>
`;
